testdisk: deduplicating testdisk/photorec

d-goldin opened the pull request:
###### Motivation for this change

Testdisk/Photorec has been packaged twice. This deduplicates
the packages by consolidating the packages into one and throwing
an error upon use of the outdated package.

this also adds qphotorec, which was previously not built and ensures
it's wrapped correctly.

Please note that I took the liberty to merge the maintainers lists.

See https://github.com/NixOS/nixpkgs/issues/67953

<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @fgaz @edolstra

d-goldin committed:
testdisk: deduplicating testdisk/photorec

Testdisk/Photorec has been packaged twice. This deduplicates
the packages by consolidating the packages into one and throwing
an error upon use of the outdated package.

this also adds qphotorec, which was previously not built and ensures
it's wrapped correctly.

Please note that I took the liberty to merge the maintainers lists.

fgaz commented on pkgs/tools/system/testdisk/default.nix:
```diff
@@ -0,0 +1,74 @@
+  ++ stdenv.lib.optional enableEwf libewf
+  ++ stdenv.lib.optional enableQt [ qtbase qttools qwt ];
+
+  nativeBuildInputs = stdenv.lib.optional enableQt [ wrapQtAppsHook ];
```
Isn't  wrapQtAppsHook only necessary when not using the qt-specific mkDerivation?

d-goldin replied:
Yep, turns out it's redundant.

fgaz commented on pkgs/tools/system/testdisk/default.nix:
```diff
@@ -0,0 +1,71 @@
+assert enableQt -> qttools != null;
+assert enableQt -> qwt != null;
+
+mkDerivation rec {
```
This brings in a qt dependency even if `enableQt = false`. It shoud be `(if enableQt then mkDerivation else stdenv.mkDerivation)`, I think

Mic92 commented on pkgs/tools/system/testdisk/default.nix:
```diff
@@ -0,0 +1,71 @@
+, ncurses
+, libuuid
+, pkgconfig
+, enableJpeg ? false, libjpeg ? null
```
Do we need so many options? Cannot users who want to disable `libjpeg` just do `testdisk.override { libjpeg = null; }`?

d-goldin replied:
Yeah, they could. For some reason I was on some explicit train at that point. I will flip at least libjpeg.

fgaz reviewed (approved):
LGTM
thatnks for the cleanup, @d-goldin

d-goldin commented:
@fgaz: Alright, thanks for the review and the good pointers. I was not very acquainted with the qt mechanisms. Let me know if I should squash or if merger could do it.

fgaz commented:
@d-goldin Well, one way or another this will need to be squashed (or the commit messages edited to match the template), so you may as well do it

Also, while I'm at it:
@GrahamcOfBorg build testdisk testdisk-qt

d-goldin commented:
Is there anything left to do before this can be merged?

vcunat commented on pkgs/top-level/all-packages.nix:
```diff
@@ -6409,7 +6409,7 @@ in
   termplay = callPackage ../tools/misc/termplay { };
 
-  testdisk-photorec = callPackage ../tools/system/testdisk-photorec { };
+  testdisk-photorec = throw "This package was a duplicate, please use testdisk or testdisk-qt instead";
```
Nitpick: perhaps use the usual `pkgs/top-level/aliases.nix`?

d-goldin replied:
@vcunat: Thanks for the pointer, I was actually not aware of this additional mechanism. Done.

Mic92 committed:
testdisk: inline platform specific option defaults

Makes it easier to read since one does not need to be aware
of overrides in all-packages.nix.

Mic92 committed:
testdisk: pkgconfig belongs to nativeBuildInputs

d-goldin commented:
Thanks.

Mic92 commented on pkgs/tools/system/testdisk/default.nix:
```diff
@@ -7,8 +7,8 @@
 , libewf
-, enableNtfs ? false, ntfs3g ? null
-, enableExtFs ? false, e2fsprogs ? null
+, enableNtfs ? stdenv.isDarwin, ntfs3g ? null
```
Oh. This should be the opposite actually.

Mic92 replied:
fixed in b731394033435002ea3f93df7bfe9d05ed6e674f
